### ISSUE TITLE
Add searching in `catalog list` command

### ISSUE THREAD

**Issue by debasishbsws** [napptive/catalog-cli#41]
#### Is your feature request related to a problem?

Right now there are so many applications in the public catalog, and even there might be so many under a single namespace. So it is quite challenging sometimes to find a particular application from the list.

Even though we have search functionalities in UI but not in the **CLI**.
![image](https://github.com/napptive/catalog-cli/assets/65381620/b4c152db-f665-47ee-a996-53073cf44d3d)

---

#### Describe the solution you'd like
In the current impl command `catalog list` returns all the app from all the namespace. It can also take `<namespace>` as arg and print all app from that particular namespace.

**My Proposal**
Instead of taking the namespace as an argument, we can take that via a flag.
for eg: `catalog list napptive` will be `catalog list -n napptive`

Additionally, for searching we can send a **Search String** as an argument, with this we can match out the application name and print only that will match.

For example: 
```bash
$./catalog list apache  # We can also add -n namespace-name  to specify a namespace
```
will print all the app that has "apache" in their name from all the namespace.
![image](https://github.com/napptive/catalog-cli/assets/65381620/dd38da11-11c7-4e60-b4d2-e7fa4189d5db)
 

#### Describe alternatives you've considered

As an alternative, we can have a completely different sub command `search` that will search for an application with the same functionality mentioned above. and we can leave the `list` command as it is.

**Comment by dhiguero** [comment]
Hi @debasishbsws, thanks for opening the issue. Just a couple of questions/comments on the approach. I think it is better to name the command `search` as it will better reflect the purpose. Do you plan to add the commands using filtering the responses on the cli side, or do you plan to add a new service on the [catalog-manager](https://github.com/napptive/catalog-manager) and call that from the CLI?

**Comment by debasishbsws** [comment]
I am planning to do it on the Cli side.
But I have seen that the admin cli commands there but have not explored that in ditails.

You can suggest to me which one will be a better approach .

**Comment by dhiguero** [comment]
Ok, Let's do the implementation incrementally :) Let's do it on the CLI side first, and in the future the team can also help with the backend implementation.

**Comment by debasishbsws** [comment]
Fixed by #42
